src/api/port.js

```diff
@@ -0,0 +1,242 @@
+import curry from 'lodash/curry';
+import flow from 'lodash/flow';
+import Immutable from 'immutable';
+import { PortRecord, PositionRecord } from '../constants/flowdesigner.model';
+import { PORT_SOURCE, PORT_SINK } from '../constants/flowdesigner.constants';
+
+const positionSelector = ['graphicalAttributes', 'position'];
+const componentTypeSelector = ['graphicalAttributes', 'portType'];
+const portTopologySelector = ['graphicalAttributes', 'properties', 'type'];
+const indexSelector = ['graphicalAttributes', 'properties', 'index'];
+
+export function isPositionRecord(position, doThrow) {
+	if (position && position instanceof PositionRecord) {
+		return true;
+	}
+	if (doThrow) {
+		throw new Error(`Should be a PositionRecord was given ${position.toString()}`);
+	}
+	return false;
+}
+
+/**
+ * Test if the first parameter is a PortRecord
+ * @param {*} port
+ * @param {bool} doThrow - throw if not a port
+ * @returns {bool}
+ * @throws
+ */
+export function isPortRecord(port, doThrow = false) {
+	if (port && port instanceof PortRecord) {
+		return true;
+	}
+	if (doThrow) {
+		throw new Error(`Should be a PortRecord was given ${port.toString()}`);
+	}
+	return false;
+}
+
+/**
+ * Test if the first parameter is a PortRecord, throw if not
+ * @param {*} port
+ * @returns {bool}
+ * @throws
+ */
+export function isPortRecordElseThrow(port) {
+	return isPortRecord(port, true);
+}
+
+/**
+ * Check if the typologie is one of the two accepted value
+ * @param {*} typology
+ * @param {bool} doThrow
+ */
+export function isTypology(typology, doThrow) {
```

Comment:
can you provide a default value to "doThrow" like you did in the previous function "isPortRecord"?